src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3773,66 +4383,82 @@ class CgroupUtils(object):
         try:
-            with open(self.cgroup_jobs_file, 'a') as f:
-                if f.tell() > 0:
-                    f.write(' ')
-                f.write(jobid)
-            return True
+            with open(self.cgroup_jobs_file, 'r+') as f:
+                joblist = f.readline().split()
+                jobset = set(joblist)
+                jobset.add(jobid)
+                f.seek(0)
+                f.write(' '.join(jobset))
+                f.truncate()
         except IOError:
             pbs.logmsg(pbs.EVENT_DEBUG, 'Failed to open cgroup_jobs file.')
+            raise
         except KeyboardInterrupt:
             raise
-        return False
 
     def remove_jobid_from_cgroup_jobs(self, jobid):
+        """
+        Remove a job idea from the file where local jobs are maintained
+        """
         pbs.logmsg(pbs.EVENT_DEBUG4,
                    'Removing jobid %s from cgroup_jobs' % jobid)
         try:
             with open(self.cgroup_jobs_file, 'r+') as f:
-                jobid_set = set(f.readline().split())
+                joblist = f.readline().split()
+                jobset = set(joblist)
                 f.seek(0)
-                jobid_set.discard(jobid)
-                f.write(' '.join(jobid_set))
+                jobset.discard(jobid)
+                f.write(' '.join(jobset))
                 f.truncate()
-            return True
         except IOError:
             pbs.logmsg(pbs.EVENT_DEBUG, 'Failed to open cgroup_jobs file.')
+            raise
         except KeyboardInterrupt:
             raise
-        return False
 
     def read_cgroup_jobs(self):
-        jobids = []
+        """
+        Read the file where local jobs are maintained
+        """
         try:
             with open(self.cgroup_jobs_file, 'r') as f:
                 jobids = f.readline().split()
         except IOError:
             pbs.logmsg(pbs.EVENT_DEBUG, 'Failed to open cgroup_jobs file.')
+            raise
         except KeyboardInterrupt:
             raise
         return jobids
 
     def delete_cgroup_jobs_file(self, jobid):
-        pbs.logmsg(pbs.EVENT_DEBUG4,
-                   'Deleting file: %s' % self.cgroup_jobs_file)
+        """
+        Delete the file where local jobs are maintained
+        """
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'Deleting file: %s' %
+                   self.cgroup_jobs_file)
         if os.path.isfile(self.cgroup_jobs_file):
             os.remove(self.cgroup_jobs_file)
-        return True
 
     def empty_cgroup_jobs_file(self):
-        pbs.logmsg(pbs.EVENT_DEBUG4,
-                   'Emptying file: %s' % self.cgroup_jobs_file)
+        """
+        Truncate the file where local jobs are maintained
+        """
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'Emptying file: %s' %
+                   self.cgroup_jobs_file)
         try:
-            with open(self.cgroup_jobs_file, 'w') as f:
+            with open(self.cgroup_jobs_file, 'w+') as f:
```

Comment:
Why w+?  You're not reading the file.

Reply:
Changed to 'w'. Thanks for all of your feedback!